scripts/Muon/thread_model.py

```diff
@@ -9,14 +10,31 @@ class ThreadModel(QThread):
     A wrapper to allow threading with
     the MaxEnt models.
     """
-    def __init__(self,model):
+    def __init__(self,model,run):
         QThread.__init__(self)
         self.model=model
+        self.run = run
 
     def __del__(self):
         self.wait()
 
+    def digit(self,x):
+        return int(filter(str.isdigit,x) or 0)
+
+    def hasDataChanged(self):
```

Comment:
I'm not sure the `TheadModel` is the correct place for this check. `ThreadModel` has the responsibility  for starting threads and it shouldn't have other responsibilities that are not related to this.

As discussed offline the `load_utils.py` might make more sense for this as that already knows about the workspace names etc.